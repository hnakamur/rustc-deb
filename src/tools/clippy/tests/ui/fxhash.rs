#![warn(clippy::default_hash_types)]
#![feature(rustc_private)]

extern crate rustc_data_structures;

use rustc_data_structures::fx::{FxHashMap, FxHashSet};
use std::collections::{HashMap, HashSet};

fn main() {
    let _map: HashMap<String, String> = HashMap::default();
    let _set: HashSet<String> = HashSet::default();

    // test that the lint doesn't also match the Fx variants themselves ğŸ˜‚
    let _fx_map: FxHashMap<String, String> = FxHashMap::default();
    let _fx_set: FxHashSet<String> = FxHashSet::default();
}
