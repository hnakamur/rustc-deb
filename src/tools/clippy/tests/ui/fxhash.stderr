error: Prefer FxHashMap over HashMap, it has better performance and we don't need any collision prevention in clippy
  --> $DIR/fxhash.rs:7:24
   |
LL | use std::collections::{HashMap, HashSet};
   |                        ^^^^^^^ help: use: `FxHashMap`
   |
   = note: `-D clippy::default-hash-types` implied by `-D warnings`

error: Prefer FxHashSet over HashSet, it has better performance and we don't need any collision prevention in clippy
  --> $DIR/fxhash.rs:7:33
   |
LL | use std::collections::{HashMap, HashSet};
   |                                 ^^^^^^^ help: use: `FxHashSet`

error: Prefer FxHashMap over HashMap, it has better performance and we don't need any collision prevention in clippy
  --> $DIR/fxhash.rs:10:15
   |
LL |     let _map: HashMap<String, String> = HashMap::default();
   |               ^^^^^^^ help: use: `FxHashMap`

error: Prefer FxHashMap over HashMap, it has better performance and we don't need any collision prevention in clippy
  --> $DIR/fxhash.rs:10:41
   |
LL |     let _map: HashMap<String, String> = HashMap::default();
   |                                         ^^^^^^^ help: use: `FxHashMap`

error: Prefer FxHashSet over HashSet, it has better performance and we don't need any collision prevention in clippy
  --> $DIR/fxhash.rs:11:15
   |
LL |     let _set: HashSet<String> = HashSet::default();
   |               ^^^^^^^ help: use: `FxHashSet`

error: Prefer FxHashSet over HashSet, it has better performance and we don't need any collision prevention in clippy
  --> $DIR/fxhash.rs:11:33
   |
LL |     let _set: HashSet<String> = HashSet::default();
   |                                 ^^^^^^^ help: use: `FxHashSet`

error: aborting due to 6 previous errors

